lib/scope.js

```diff
@@ -174,17 +174,19 @@ Scope.prototype.buildFilter = function buildFilter() {
 
   if (arguments[0] instanceof RegExp) {
     return function(candidate) {
-      // TODO-coverage: Try to find an actual use case where a non-string
-      // argument can be passed here. Don't add coverage that uses '' which
-      // technically would cover this block, but would not be helpful. If
-      // a use case can't be found, remove the conditional.
-      if (candidate) {
-        candidate = candidate.replace(
-          filteringArguments[0],
-          filteringArguments[1]
+      /* istanbul ignore if */
+      if (typeof candidate !== 'string') {
+        // Given the way nock is written, it seems like `candidate` will always
+        // be a string, regardless of what options might be passed to it.
+        // However the code used to contain a truthiness test of `candidate`.
+        // The check is being preserved for now.
+        //
+        // If you DO encounter this error, please report it as a bug.
+        throw Error(
+          `Nock internal assertion failed: typeof candidate is ${typeof candidate}`
```

Comment:
Maybe do the " you DO encounter this error, please report it as a bug" to the error message?